src/Makefile.am

```diff
@@ -184,6 +186,29 @@ signallist.i: $(glade_file) Makefile
 
 CLEANFILES += signallist.i
 
+# embed files using gresource
+resource_dependencies = geany.gresource.xml $(top_srcdir)/data/geany.glade
+resources.c: $(resource_dependencies)
+	$(AM_V_GEN)$(GLIB_COMPILE_RESOURCES) \
+		--target=$@ \
+		--sourcedir=$(srcdir) \
+		--generate-source \
+		--c-name="geany" \
+		--manual-register \
+		--internal \
+			$(srcdir)/geany.gresource.xml
+resources.h: $(resource_dependencies)
+	$(AM_V_GEN)$(GLIB_COMPILE_RESOURCES) \
+		--target=$@ \
+		--sourcedir=$(srcdir) \
+		--generate-header \
+		--c-name="geany" \
+		--manual-register \
+		--internal \
+			$(srcdir)/geany.gresource.xml
```

Comment:
Hum, maybe you need to add something like `|| rm -f $@` (but you need to still return in error) so that if the generation fails for some reason it will be tried again next time.  Another common solution is to generate a temporary file and move it over on success, so it's mostly atomic.

Reply:
Are you sure this tool is too stupid to not properly update the file (or not) depending on the outcome?

Reply:
I'm not sure, but many tools aren't smart enough so I'd at least check.

Reply:
This runs glib-compile-resources twice. It's usually better to a construct pattern rule, where (GNU) make recognizes that one rule produces multiple output files. That's assuming it's even possible with, I haven't checked.

```
%.gresource.c %.gresource.h: %.gresource.xml $(top_srcdir)/data/geany.glade
[…]
```

Reply:
@kugel- It's supposed to run it twice, once for the source file (`--generate-source`) and once for the header file (`--generate-header`).